src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs

```diff
@@ -29,6 +31,11 @@ public void Dispose()
 
     public async ValueTask DisposeAsync()
     {
+        if (fileStream is null)
+        {
+            return;
+        }
```

Comment:
In this context we know dispose is never invoked multiple times OR ran in parallel but usually the pattern is to ensure `isDisposed` is set ASAP to prevent more-than-once invocation of the dispose methods.

For readability I would suggest:
```suggestion
		var isDisposed = fileStream is null;
        if (isDisposed)
        {
            return;
        }
```

The rest of the method isn't great but its existing code and not really bad either...